lib/rack_graphql/middleware.rb

```diff
@@ -74,7 +74,10 @@ def post_request?(env)
     end
 
     def post_data(env)
-      ::Oj.load(env['rack.input'].gets.to_s)
+      payload = env['rack.input'].gets.to_s
```

Comment:
Won't this check for the presence of the `\0` character in the first line? JSON can contain newline characters (especially if pretty-printed). I'd use `.read` here.

Reply:
good catch @psyho ! Updated to .read

Reply:
> Won't this check for the presence of the \0 character in the first line?

No, rack gql input has one line only and new line chars are present in the query payload and are escaped, e.g.
`"{\"query\":\"query {\\n  health\\n}\",\"variables\":null}"`